src/agent/communicator/include/communicator.hpp

```diff
@@ -13,30 +16,22 @@ namespace communicator
     class Communicator
     {
     public:
-        Communicator();
+        Communicator(const std::shared_ptr<iConfigurationParser>& configParser);
```

Comment:
I think we can handle this without any type of reference to an external class, at least for the time being. It just needs to get a function that takes two strings and returns one.

Reply:
Made changes according to our conversation.